data-loader/core/src/main/java/com/scalar/db/dataloader/core/DataLoaderError.java

```diff
@@ -0,0 +1,292 @@
+  DUPLICATE_DATA_MAPPINGS(
+      Category.USER_ERROR,
+      "0017",
+      "Duplicate data mappings found for table '%s' in the control file",
+      "",
+      ""),
+  MISSING_COLUMN_MAPPING(
+      Category.USER_ERROR,
+      "0018",
+      "No mapping found for column '%s' in table '%s' in the control file. Control file validation set at 'FULL'. All columns need to be mapped",
+      "",
+      ""),
+  CONTROL_FILE_MISSING_DATA_MAPPINGS(
+      Category.USER_ERROR, "0019", "The control file is missing data mappings", "", ""),
+  TARGET_COLUMN_NOT_FOUND(
+      Category.USER_ERROR,
+      "0020",
+      "The target column '%s' for source field '%s' could not be found in table '%s'",
+      "",
+      ""),
+  MISSING_PARTITION_KEY(
+      Category.USER_ERROR,
+      "0021",
+      "The required partition key '%s' is missing in the control file mapping for table '%s'",
+      "",
+      ""),
+  MISSING_CLUSTERING_KEY(
+      Category.USER_ERROR,
+      "0022",
+      "The required clustering key '%s' is missing in the control file mapping for table '%s'",
+      "",
+      ""),
+  MULTIPLE_MAPPINGS_FOR_COLUMN_FOUND(
+      Category.USER_ERROR,
+      "0023",
+      "Duplicated data mappings found for column '%s' in table '%s'",
+      "",
+      ""),
+  MISSING_CLUSTERING_KEY_COLUMN(
+      Category.USER_ERROR,
+      "0024",
+      "Missing required field or column mapping for clustering key %s",
+      "",
+      ""),
+  MISSING_PARTITION_KEY_COLUMN(
+      Category.USER_ERROR,
+      "0025",
+      "Missing required field or column mapping for partition key %s",
+      "",
+      ""),
+  MISSING_COLUMN(Category.USER_ERROR, "0026", "Missing field or column mapping for %s", "", ""),
+  VALUE_TO_STRING_CONVERSION_FAILED(
+      Category.USER_ERROR,
+      "0027",
+      "Something went wrong while converting the ScalarDB values to strings. The table metadata and Value datatype probably do not match. Details: %s",
+      "",
+      ""),
+  FILE_FORMAT_NOT_SUPPORTED(
+      Category.USER_ERROR, "0028", "The provided file format is not supported : %s", "", ""),
+  COULD_NOT_FIND_PARTITION_KEY(
+      Category.USER_ERROR, "0029", "Could not find the partition key", "", ""),
+  UPSERT_INSERT_MISSING_COLUMNS(
+      Category.USER_ERROR,
+      "0030",
+      "The source record needs to contain all fields if the UPSERT turns into an INSERT",
+      "",
+      ""),
+  DATA_ALREADY_EXISTS(Category.USER_ERROR, "0031", "Record already exists", "", ""),
+  DATA_NOT_FOUND(Category.USER_ERROR, "0032", "Record was not found", "", ""),
+  COULD_NOT_FIND_CLUSTERING_KEY(
+      Category.USER_ERROR, "0033", "Could not find the clustering key", "", ""),
+  TABLE_METADATA_MISSING(Category.USER_ERROR, "0034", "No table metadata found", "", ""),
+  MISSING_SOURCE_FIELD(
+      Category.USER_ERROR,
+      "0035",
+      "The data mapping source field '%s' for table '%s' is missing in the json data record",
```

Comment:
I think `json` be capitalized here since we're talking about the format.

```suggestion
      "The data mapping source field '%s' for table '%s' is missing in the JSON data record",
```

Or, if we're referring to the file extension, it should be as follows:

```suggestion
      "The data mapping source field '%s' for table '%s' is missing in the .json data record",
```